### ISSUE TITLE
SameSite value of 'None' not supported

### ISSUE THREAD

**Issue by jsurls** [pallets/werkzeug#1549]
Google's recent [blog post](https://blog.chromium.org/2019/05/improving-privacy-and-security-on-web.html
) describes changes on how they will enforce Cookies with SameSite value. The referenced [samesite cookies explained article](https://web.dev/samesite-cookies-explained/) and [RFC](https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03#section-4.1) both describe the ability to specify a SameSite value of 'None' the string (as opposed to None, not set) for SameSite.

The following today produces an error stating _invalid SameSite value; must be 'Strict', 'Lax' or None_
```python
resp.set_cookie('somekey', 'someval', domain='.mydomain.com', samesite='None')
```

You would think setting ```None``` would work but instead it unsets the key and prevents the samesite attribute from being present with a None (string) value in the headers.
```python
resp.set_cookie('somekey', 'someval', domain='.mydomain.com', samesite=None)
```

**Comment by davidism** [comment]
I'm always hesitant when an RFC is still in draft, or a feature is only supported by some browsers. The article states that browsers that don't understand `None` should ignore it, but does this actually happen? Would be good to check before allowing inconsistent behavior.

**Comment by jsurls** [comment]
That's a fair point. It's still early but I suspect a blog declaration like the above indicates it's likely going to see daylight. The only reason I ran into this was because I was attempting to test the implications of the announcement in current Chrome vs Chromium (which has the changes).

I opened a PR #1550 since it was a simple fix but I understand if you think it's better to hold off until the RFC is past a draft state.

I'm assuming other developers will want to easily test the 'None' too and the PR I supplied should provide a quick guide on how to manually tweak it locally to test this out.

**Comment by davidism** [comment]
Well, turns out the entire `SameSite` feature is a draft spec, so I'm ok with this in 1.0.

**Comment by dlesco** [comment]
@davidism : Chrome is implementing new SameSite=None security in February, and other browser makers also intend to implement the new security model:

https://blog.chromium.org/2019/10/developers-get-ready-for-new.html

Will Werkzeug 1.0, with this fix, be released very soon?  Or should the fix be backported to the 0.x series?

**Comment by dlesco** [comment]
Created new issue #1660 on getting this fix released.

**Comment by homm** [comment]
> Please do not create these types of issues, they do not make the release process happen faster.

Ok.

Several months ago, issue #1549 fixed the SameSite='None' problem. However, it was never released, it's scheduled for 1.0.

Meanwhile, Chrome says their changing their security model in **February** (in 20 days), and they said other browser makers are doing the same:

https://blog.chromium.org/2019/10/developers-get-ready-for-new.html

We all need to start incorporating this into a web service. Will 1.0, with this fix, be released very soon, or should this be backported to the 0.x series?

**Comment by pgjones** [comment]
For those following this issue please could you test Werkzeug 1.0.0 release candidate 1, 1.0.0rc1, this release supports this feature and if the release candidate is good we can release 1.0.0.
